### PR TITLE
Theme selectors improvements

### PR DESCRIPTION
**Pull request by aditudorache**
## Amsterdam styled components pull request

Before opening a pull request, please ensure:

- [x] The default export from a file matches the file name
- [x] The component styles are only placed in the `asc-core/src/components` folder. Exception is the `asc-ui/src/internals` folder where the styles are allowed. The components from this folder are just for internal use in the stories.
- [x] The component style name follows the pattern `<ComponentName>Style/<ComponentName>Style.ts`
- [x] Pull request has tests (we are going for 100% coverage!)

Be kind to code reviewers, please try to keep pull requests as small and focused as possible :)

### CODE REVIEW

File: packages/asc-core/src/utils/themeUtils.ts [hunk 1]
```diff
@@ -20,39 +29,42 @@ export const getTypographyFromTheme = (
...
   })
-  return breakpoint && breakpoint(type)
+  return breakpointFunc && breakpointFunc(type)
 }
 
-export const fillSvgFromTheme = (
-  theme: Theme.ThemeInterface,
+export const svgFill = (
   colorType?: Theme.TypeLevel,
   variant: string = 'main',
```
Comment (chrisvanmook) [inline packages/asc-core/src/utils/themeUtils.ts]: we can type the variants, as we know which ones are restricted (cannot be only just any string). perhaps we can use keyof here?

File: packages/asc-core/src/utils/themeUtils.ts [hunk 2]
```diff
@@ -3,6 +3,15 @@ import { css } from '../styled-components'
 import { Theme } from '../theme'
 import { fromTheme } from '.'
 
+export const color = (
+  colorType?: Theme.TypeLevel,
+  variant: string = 'main',
```
Comment (chrisvanmook) [inline packages/asc-core/src/utils/themeUtils.ts]: we can type the variants, as we know which ones are restricted (cannot be only just any string). perhaps we can use keyof here?

File: packages/asc-core/src/components/ButtonStyle/ButtonStyle.ts [hunk 3]
```diff
@@ -25,7 +25,7 @@ const ButtonStyle = styled(ButtonBaseStyle)<Props>`
   & svg {
     width: 30px;
     height: 30px;
-    ${({ color, theme }) => fillSvgFromTheme(theme, color)};
+    ${({ color, theme }) => svgFill(color)({ theme })};
```
Comment (chrisvanmook) [inline packages/asc-core/src/components/ButtonStyle/ButtonStyle.ts]: plz find a clever way to curry theme to svgFill

File: packages/asc-core/src/utils/themeUtils.ts (hunk 2 above)
Comment (aditudorache) [reply packages/asc-core/src/utils/themeUtils.ts]: will be done in the second iteration

File: packages/asc-core/src/utils/themeUtils.ts (hunk 1 above)
Comment (aditudorache) [reply packages/asc-core/src/utils/themeUtils.ts]: will be done in the second iteration

File: packages/asc-core/src/components/ButtonStyle/ButtonStyle.ts (hunk 3 above)
Comment (aditudorache) [reply packages/asc-core/src/components/ButtonStyle/ButtonStyle.ts]: will be done in the second iteration

**Pull request merged**